File: SPRING-VSC/WorksPace-Cook/cook/src/main/java/com/mx/cook/service/ProductoService.java
package com.mx.cook.service;

import java.util.List;

import com.mx.cook.dto.ProductoDTO;

public interface ProductoService {

    List<ProductoDTO> listarProducto();

    List<ProductoDTO> listarProductoDisponibles();

    List<ProductoDTO> listarProductoPorCategoria(int categoriaId);

    ProductoDTO buscarProducto(int id);

    void guardarProducto(ProductoDTO productoDTO);

    void editarProducto(ProductoDTO productoDTO);

    void eliminarProducto(int id);

}

File: ANGULAR-VSC/WoksPace-Cook/cook-front/src/app/services/producto.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Producto } from "../models/producto";

@Injectable({
    providedIn: 'root'
})
export class ProductoService {
    private readonly baseUrl = 'http://localhost:8081/api/productos';

    constructor(private readonly http: HttpClient) { }

    listarProductos(): Observable<Producto[]> {
        return this.http.get<Producto[]>(`${this.baseUrl}/listarProducto`);
    }

    listarDisponibles(): Observable<Producto[]> {
        return this.http.get<Producto[]>(`${this.baseUrl}/productoDisponibles`);
    }

    listarPorCategoria(categoriaId: number): Observable<Producto[]> {
        return this.http.get<Producto[]>(`${this.baseUrl}/listarProductoCategoria/${categoriaId}`);
    }

    buscarProducto(id: number): Observable<Producto> {
        return this.http.get<Producto>(`${this.baseUrl}/buscarProducto/${id}`);
    }

    guardarProducto(producto: Producto): Observable<void> {
        return this.http.post<void>(`${this.baseUrl}/guardarProducto`, producto);
    }

    editarProducto(producto: Producto): Observable<void> {
        return this.http.put<void>(`${this.baseUrl}/editarProducto`, producto);
    }

    eliminarProducto(id: number): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/eliminarProducto/${id}`);
    }
}
